### PR TITLE
fix linking Libdvd (fixes: #16932)

### PR DESCRIPTION
**Pull request by wsnipex**
This should fix the missing symbols in libdvd reported in http://trac.kodi.tv/ticket/16932 when building libdvd with cmake.

It also fixes double building libdvd on unified depends platforms, by first checking if depends already have the libs, but I didn't have time to really test this case.

@FernetMenta @Memphiz

### CODE REVIEW

**Comment by FernetMenta** [comment]
thanks bob for keeping my inbox full :)

**Comment by Memphiz** [comment]
Mhh fernet identified something in configure.ac of libdvdread to be the problem - is this PR really complete? (Not seeing a change to anything configure.ac related)

**Comment by FernetMenta** [comment]
@wsnipex thanks. I can test tomorrow evening in case you have not done so far.
@Memphiz I think to remember that I didn't have dvd_reader in my build tree yesterday. Not sure if that is still required. Maybe I was wrong.

**Comment by wsnipex** [comment]
it works for me on linux without unified depends. Cannot test MacOS and friends.
The issue has nothing to do with libdvdread's configure.
The whole wraplib stuff is a bit of a hack, we link those with -Wl,--unresolved-symbols=ignore-all so we have no idea if we did something wrong with building until runtime. In this case we didn't link dvdnav against dvdread, so all its symbols were missing.

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
if it works on Linux, my comments from yesterday are obsolete.

**Comment by lrusak** [comment]
I'm having an issue with this on LibreELEC.

first, the pkg-config files seems to be wonky, not sure if that's use or just how the libdvd* packages are.
```
toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/lib/pkgconfig/dvdnav.pc
toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/lib/pkgconfig/dvdread.pc
toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/lib/pkgconfig/libdvdcss.pc
```

but even when changing libdvdnav --> dvdnav and libdvdread --> dvdread, the build still bails out with lots of this
```
<snip>
nav_read.o (symbol from plugin): In function `navRead_PCI':
(.text+0x0): multiple definition of `navRead_PCI'
nav_read.o (symbol from plugin):(.text+0x0): first defined here
nav_read.o (symbol from plugin): In function `navRead_PCI':
(.text+0x0): multiple definition of `navRead_DSI'
nav_read.o (symbol from plugin):(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
CMakeFiles/wrap_libdvdnav.dir/build.make:60: recipe for target 'system/players/VideoPlayer/libdvdnav-x86_64.so' failed
make[2]: *** [system/players/VideoPlayer/libdvdnav-x86_64.so] Error 1
```

Here is my CMakeCache.txt, http://sprunge.us/SaGA

Not sure if it's our sysroot paths or the fact we use static libs or what.

File: project/cmake/modules/FindLibDvd.cmake [hunk 1]
```diff
@@ -4,154 +4,191 @@ if(ENABLE_DVDCSS)
...
+ 
+    find_path(LIBDVD_INCLUDE_DIR dvdcss/dvdcss.h PATHS ${PC_DVD_INCLUDE_DIRS})
+    find_library(DVDREAD_LIBRARY NAMES dvdread libdvdread PATHS ${PC_DVD_libdvdread_LIBDIR})
+    find_library(DVDNAV_LIBRARY NAMES dvdnav libdvdnav PATHS ${PC_DVD_libdvdnav_LIBDIR})
+    find_library(DVDCSS_LIBRARY NAMES dvdcss libdvdcss PATHS ${PC_DVD_libdvdcss_LIBDIR})
+
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(LIBDVD REQUIRED_VARS LIBDVD_INCLUDE_DIR DVDREAD_LIBRARY DVDNAV_LIBRARY DVDCSS_LIBRARY)
+    if(LIBDVD_FOUND)
+      set(LIBDVD_LIBRARIES ${DVDREAD_LIBRARY} ${DVDNAV_LIBRARY} ${DVDREAD_LIBRARY})
```
Comment (lrusak) [inline project/cmake/modules/FindLibDvd.cmake]: should one of these be `DVDCSS_LIBRARY`?

Comment (wsnipex) [inline project/cmake/modules/FindLibDvd.cmake]: yes, already fixed locally. Currently test building

Comment (lrusak) [inline project/cmake/modules/FindLibDvd.cmake]: making this line
```
set(LIBDVD_LIBRARIES ${DVDREAD_LIBRARY} ${DVDCSS_LIBRARY})
```
allows it to compile for me. seems the double ${DVDNAV_LIBRARY} in the `core_link_library` causes problems.

**Comment by wsnipex** [comment]
should be fixed
jenkins build this please

**Comment by lrusak** [comment]
I can confirm this now builds and runs :+1:

**Comment by wsnipex** [comment]
meh, ofc ios has to be a pain again
jenkins build this please

**Comment by wsnipex** [comment]
jenkins build this please

@Memphiz any reason ios builds without dvdcss?

**Comment by Memphiz** [comment]
Dvdcss didn't work on iOS - i did a pr with dvdcss Support for iOS but i didn't merge it because there was no demand for it - do its expected to not use dvdcss on iOS

**Comment by wsnipex** [comment]
can you test if the issue is fixed now on darwin? If yes, press the button

**Comment by Memphiz** [comment]
I can't really test this atm ... Alotta trouble with the new flat :(

**Comment by Memphiz** [comment]
But someone can do a test build for osx and give it to the user that filed the track ticket...

**Comment by FernetMenta** [comment]
I'll do a test on my machine today and merge if ok.

**Comment by Memphiz** [comment]
Would be Great :)

**Pull request merged**